python/tank/bootstrap/manager.py

```diff
@@ -60,22 +61,110 @@ def __repr__(self):
         repr += " Base %s >" % self._base_config_descriptor
         return repr
 
-    def _set_pipeline_configuration(self, name):
+    def _get_pipeline_configuration(self):
         """
         The pipeline configuration that is being operated on.
         By default, the primary pipeline config will be used.
         """
-        self._pipeline_configuration_name = name
+        return self._pipeline_configuration_name
 
-    def _get_pipeline_configuration(self):
+    def _set_pipeline_configuration(self, name):
         """
         The pipeline configuration that is being operated on.
         By default, the primary pipeline config will be used.
         """
-        return self._pipeline_configuration_name
+        self._pipeline_configuration_name = name
 
     pipeline_configuration = property(_get_pipeline_configuration, _set_pipeline_configuration)
 
+
+    def _get_do_shotgun_config_lookup(self):
+        """
+        Flag to indicate if the bootstrap process should connect to
+        Shotgun and attempt to resolve a config. Defaults to True.
+
+        If ``True``, the bootstrap process will connect to Shotgun as part
+        of the startup, look for a pipeline configuration and attempt
+        to resolve a toolkit environment to bootstrap into via the
+        Pipeline configuration data. Failing this, it will fall back on
+        the :meth:`base_configuration`.
+
+        If ``False``, no Shotgun lookup will happen. Instead, whatever config
+        is defined via :meth:`base_configuration` will always be used.
+        """
```

Comment:
Do we need the docstring for both the set and get?  For sphinx won't one do?  I'm worried that if we go in and update things we'll need to keep these in sync and it'll get missed if we dup it.